test/performance/mako/sidecar.go

```diff
@@ -16,9 +16,38 @@ limitations under the License.
 
 package mako
 
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/mako/helpers/go/quickstore"
+	qpb "github.com/google/mako/helpers/proto/quickstore/quickstore_go_proto"
+	"knative.dev/pkg/changeset"
+)
+
 const (
-	// SidecarAddress is the address of the Mako sidecar to which we locally
+	// sidecarAddress is the address of the Mako sidecar to which we locally
 	// write results, and it authenticates and publishes them to Mako after
 	// assorted preprocessing.
-	SidecarAddress = "localhost:9813"
+	sidecarAddress = "localhost:9813"
 )
+
+// SetupMako sets up the mako client for the provided benchmarkKey.
+// It will add a few common tags and allows each benchmark to add custm tags as well.
+// It returns the mako client handle to sotre metrics, a method to close the connection
+// to mako server once done and error if case of failures.
+func Setup(ctx context.Context, benchmarkKey *string, extraTags ...string) (*quickstore.Quickstore, func(context.Context), error) {
+	var close func(context.Context)
```

Comment:
why is this variable required?

Reply:
I need to return an empty function when errors occur and I found it best to create it once and use it for all the errors.

Reply:
Just return nil?

Reply:
yeah, just nil. Nobody's gonna call it anyway